docs/testing-styleguide.MD

```diff
@@ -0,0 +1,280 @@
+
+Where the stories in the main `Design System` documentation directory do not cover all scenarios required to fully test the component, additional stories (named `basic`/`grouped`/`styles overridden` etc.) must be created in the `Test` directory. For example, scenarios testing special characters, all combinations of size/type/state etc. should have stories in the `Test` directory under the relevant component. Stories for visual testing should also be located here and named `visual`. These stories should be written with the components in set states where possible to avoid the use of knobs although they may be required in some cases, like to allow different text strings to be entered for instance. A developer may need to assist with writing the additional required stories here.
+
+To clarify the purpose of each directory,
+
+| Directory     | Usage                                                                    |
+|---------------|--------------------------------------------------------------------------|
+| Design System | User facing documentation, Cypress regression tests                      |
+| Test          | Additional stories for Cypress regression tests, visual regression tests |
+
+#### Automated Testing
+All regression, accessibility, theme and visual tests are written using the [Cypress.io](https://www.cypress.io) framework with all knob configurations (where applicable) and events tested, along with accessibility via the [Axe](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd) add on. Instructions of how to install Cypress can be found in the [Getting Started](https://github.com/Sage/carbon/blob/master/cypress/README.md) guide.
+
+Where functionality is already tested in unit testing, this does not need to be repeated in assertions in Cypress tests, with the exception of events tests which should be repeated here in order to test actions in a manner more closely resembling that of a user in a browser.
+
+The visual comparison testing tool [Applitools](https://applitools.com) is used to test for UI regressions at each build by comparing new snapshots of the component preview iFrame with previous baseline ones. Applitools Eyes commands are included in Cypress tests where required. Visual test files should be stored in the `/visual` directory.
+
+##### Cypress File Structure
+```
+.
+├── cypress
+│ ├── fixture
+│ ├── features
+│ │   ├── regression
+│ │       ├── accessibility
+│ │       ├── designSystem
+│ │       ├── experimental
+│ │       ├── test
+│ │       ├── themes
+│ │       ├── validations
```

Comment:
Please delete this line, because all `validation` tests were moved to the `visual` and deleted from cypress `regression` suite.

`validations`

Reply:
I took this straight from the original Cypress README.md so assumed everything on there was current information.

Reply:
I'll update this document once the changes are in `master`.